pkg/component/watch.go

```diff
@@ -39,29 +58,28 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 		return nil
 	}
 
+	// If path to be watched is directory, walk the directory recursively ignoring all ignores path and paths under them and accumulate list of all such(eligible) files
 	folders := []string{}
 	err = filepath.Walk(path, func(newPath string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-
-		if info.IsDir() {
+		// If the src root is watched, automatically all files and folders under it will be watched immaterial of contents of ignores.
+		// So, inorder to avoid the above, src root dir is not watched
+		// Watch the regular(non-dir) files whose parents are not already watched => watch on the src root dir's non-folder files
+		if parent := filepath.Dir(newPath); !stringinSlice(parent, folders) && !info.IsDir() && !isIgnore(newPath, ignores) {
 			folders = append(folders, newPath)
```

Comment:
this is weird. 
If I understand it correctly it adds regular files to the variable called `folders` :-(

Reply:
restoring it now...
Had some misunderstanding about addwatcher it seems addwatch on a directory watches only everything immediately under it and is most essential to track any creations/deletions within a directory while only updates can be caught using watch on files...
So restored the old code with a slight change to fix https://github.com/redhat-developer/odo/issues/639